pages/product.page.ts

```diff
@@ -0,0 +1,37 @@
+import { Locator, Page } from "@playwright/test";
+import { HeaderFragment } from './fragments/header.fragment';
+
+
+export class ProductPage{
+    page: Page;
+    header: HeaderFragment;
+    productImg: Locator;
+    productName: Locator;
+    productPrice: Locator;
+    productDescr: Locator;
+    addToCartBtn: Locator;
+    addToFavouritesBtn: Locator;
+    decreaseQntty: Locator;
+    productQntty: Locator;
+    increaseQntty: Locator;
+
+
+    constructor(page: Page){
+        this.page = page;
+        this.header = new HeaderFragment(page);
+        this.productImg = this.page.locator('.figure-img.img-fluid');
+        this.productName = this.page.getByTestId('product-name');
+        this.productPrice = this.page.getByTestId('unit-price');
+        this.productDescr = this.page.getByTestId('product-description');
+        this.addToCartBtn = this.page.getByTestId('add-to-cart');
+        this.addToFavouritesBtn = this.page.getByTestId('add-to-favorites');
+        this.decreaseQntty = this.page.getByTestId('decrease-quantity');
+        this.productQntty = this.page.getByTestId('quantity');
+        this.increaseQntty = this.page.getByTestId('increase-quantity');
+    }
+
+
+    returnHashtagLocator(text: string): Locator{
+        return this.page.locator('.badge.rounded-pill').filter({has: this.page.getByText(text)});
```

Comment:
Judging from your first version of this locator, the filter is should check the text of this same element (the .badge.rounded-pill), not a child element inside it.
In this case, you can simplify it by using hasText instead of has, like this:
```suggestion
        return this.page.locator('.badge.rounded-pill').filter({ hasText: text });
```
This is shorter and easier to read because hasText is made for matching the text content of the element itself.